src/main/java/com/spartronics4915/frc2020/subsystems/Climber.java

```diff
@@ -17,50 +19,45 @@
  */
 public class Climber extends SpartronicsSubsystem
 {
-    private TalonSRX mLiftMotor;
-    private CANSparkMax mWinchMotor;
+    private final SpartronicsMotor mLiftMotor;
+    private final SpartronicsMotor mWinchMotor;
 
     public Climber()
     {
         // Hardware Contructor (Add motors and such here when I get them)
-        mLiftMotor = new TalonSRX(Constants.Climber.kLiftMotorId);
-        mWinchMotor = new CANSparkMax(Constants.Climber.kWinchMotorId, MotorType.kBrushless);
+        mLiftMotor = SpartronicsSRX.makeMotor(Constants.Climber.kLiftMotorId, SensorModel.fromMultiplier(1));
+        mWinchMotor = SpartronicsMax.makeMotor(Constants.Climber.kWinchMotorId, SensorModel.fromMultiplier(1));
     }
 
     public void extend()
     {
-        mLiftMotor.set(ControlMode.PercentOutput, Constants.Climber.kExtendSpeed);
-        mWinchMotor.set(0.0);
+        mLiftMotor.setDutyCycle(Constants.Climber.kExtendSpeed);
+        mWinchMotor.setDutyCycle(0.0);
     }
 
     public void winch(boolean stalled)
     {
-        mLiftMotor.set(ControlMode.PercentOutput, 0.0);
+        mLiftMotor.setDutyCycle(0.0);
         if (stalled)
-            mWinchMotor.set(Constants.Climber.kWinchSpeed);
+            mWinchMotor.setDutyCycle(Constants.Climber.kWinchSpeed);
         else
-            mWinchMotor.set(-Constants.Climber.kWinchSpeed);
+            mWinchMotor.setDutyCycle(Constants.Climber.kReverseWinchSpeed);
     }
 
     public void retract()
     {
-        mLiftMotor.set(ControlMode.PercentOutput, -Constants.Climber.kExtendSpeed);
-        mWinchMotor.set(0.0);
+        mLiftMotor.setDutyCycle(Constants.Climber.kReverseExtendSpeed);
+        mWinchMotor.setDutyCycle(0.0);
     }
 
     public void stop()
     {
-        mLiftMotor.set(ControlMode.PercentOutput, 0.0);
-        mWinchMotor.set(0.0);
-    }
-
-    public double getWinchVoltage()
-    {
-        return mWinchMotor.getBusVoltage();
+        mLiftMotor.setDutyCycle(0.0);
+        mWinchMotor.setDutyCycle(0.0);
     }
 
     public boolean isStalled()
     {
-        return false;
+        return mWinchMotor.getOutputCurrent() == 10;
```

Comment:
hm - i would think:

a) this hard-coded constant should be in Constants
b) that it shouldn't be  ==, but rather >=

Reply:
I also note that in RobotContainer:

    new JoystickButton(mButtonBoard, 10).whileHeld(mClimberCommands.new Extend(mClimber));

If you use mClimberCommands.new, you shouldn't the need mClimber since its a member variable of ClimberCommands.

Also:

the use of chaining (via .andThen) is a cool way to produce a command group.  My concern is merely that this logic probably belongs in ClimberCommands and not RobotContainer.  This general rule of encapsulation:  "keep details as private as possible" is the secret of programming success.